src/NLog/Fluent/LogBuilder.cs

```diff
@@ -294,36 +294,28 @@ public LogBuilder StackTrace(StackTrace stackTrace, int userStackFrame)
             if (!_logger.IsEnabled(_logEvent.Level))
                 return;
 
-            // TODO NLog ver. 5 - Remove these properties
-            if (callerMemberName != null)
-                Property("CallerMemberName", callerMemberName);
-            if (callerFilePath != null)
-                Property("CallerFilePath", callerFilePath);
-            if (callerLineNumber != 0)
-                Property("CallerLineNumber", callerLineNumber);
-
-            _logEvent.SetCallerInfo(null, callerMemberName, callerFilePath, callerLineNumber);
+            SetCallerInfo(callerMemberName, callerFilePath, callerLineNumber);
 
             _logger.Log(_logEvent);
         }
 #else
-        /// <summary>
-        /// Writes the log event to the underlying logger.
-        /// </summary>
-        public void Write()
+        public void Write(
+            string callerMemberName = null,
+            string callerFilePath = null,
+            int callerLineNumber = 0)
```

Comment:
I don't fully understand why the signature changed for `Write` for < net 45? 

(the docs outside is a good idea)

Reply:
I want to make the Logger-interface binary compatible.

Reply:
While that's a nice property, it's not really useful for now? We can't (runtime) replace the nlog.dll for .net4 with .net45 already. 

I prefer revert of this as it could confuse users. (the optional parameters are visible)

Reply:
Not sure I understand the issue with confused users. Net45 users are also looking at default parameters.

If you have a net40 library compiled with NLog Net40 that uses fluent-interface, and then you attempt to use that library in an Net45 application, then you will get method-not-found. This is the issue I'm trying to fix. The NLog Logger-interface should not give such surprises.

Reply:
> This is the issue I'm trying to fix

ah I would expect more issues, but OK you convinced me.  :)